moveit_ros/moveit_servo/demos/cpp_interface/demo_joint_jog.cpp

```diff
@@ -0,0 +1,105 @@
+/*******************************************************************************
```

Comment:
These `demo` files as they are would actually serve as good integration tests. Basically do something like:

* Start node
* Record the joint states
* Jog the 7th joint for a while
* Check that the final joint states are different

Unsure if the timing will be accurate enough to actually check the value of the final joint state (probably not), but having it move at least validates the interface

Reply:
That is a good idea.
Maybe we can do an Eigen `isApprox` check with a slightly bigger tolerance for where we expect it to be.